### ISSUE TITLE
The Mod Is... Kind Of A Mess

### ISSUE THREAD

**Issue by Kalman98** [TEAMModding/KalStuff#28]
So, I don't really want to write this, but I feel like we should talk about it to decide what to do next with the mod. Pretty much, it feels like kind of a mess to me. Don't get me wrong, I'm not trying to say that anyone's work is bad. It's just that a lot of it doesn't seem to fit in. I think the biggest example of this is sodas. They are a neat item and all... but where do you get the fancy cans and the carbonation? As @pianoman373 has suggested before we could add a carbonation/soda machine, but that really doesn't fit the theme of Minecraft at all.

We also have jewel and blaze soup. Honestly, I don't know what to think of these. Isn't a strength potion the same as blaze soup? Except strength potions don't give you multiple super OP potion effects. Jewel soup is... eh. First, why are you eating a diamond? That has to hurt **badly** going through your system. Second, Speed II is way cheaper to come by normally. Regeneration III doesn't even exist in vanilla survival, but Regeneration II is way cheaper as well. I think giving it more over powered effects would just mess up vanilla potions even more, though.

We also have trash cans. Those look neat and all, but honestly, there are dozens of mods out there that add them, and this trash can is as basic as it gets. Unless we come up with some super unique ideas for it, I don't see any reason to have it in the mod.

Food blocks. In my opinion these look awesome, but I'm not entirely sure why we added them in the first place. I don't know if we should keep them or get rid of them, they aren't really hurting anything as far as I can think of. It just seems odd to me that you can only compress those three foods.

Chicken nests. I know I am the one that added these anyway, but I can't figure out a decent purpose for them. The only thing they do is make egg farming more humane, and most people don't even care about that. It's way more efficient to drop some chickens in a tiny hole over a hopper than it is to spread them all out over chicken nests. I'm thinking maybe these should be removed.

A few other questionable items are moon flowers, grapes, wine, and teas. Moon flowers are just cool, and add a little more atmosphere to the world. Eventually they could serve a purpose as well (luminescent paint, anyone?) Grapes are a plain food item, but they are necessary to craft wine. And wine, well, I have big plans for that, I just haven't gotten around to them yet, as they are... big. Likewise, I have quite a few plans for the tea, I just haven't worked on them yet either.

Oh, and I know it's been mentioned that crushed ender should be removed before. I don't think that we should remove our item just because Mojang copied us and tweaked a few things. Crushed ender is still better than chorus fruit, and I think it's cool that we did it before they did.

So now, what are everyone's thoughts? Do you think we should remove some items? Have some suggestions for tweaks to make them fit in better? Both? Feel free to disagree with me as well. We can discuss it all here, and once we've decided on things we can go ahead and do whatever they are. Like I said, I don't think anyone's work is bad, a lot of it just doesn't fit in with vanilla or the rest of the mod.

If you read this far, congratulations! Now let's get this mod cleaned up. :smiley:

**Comment by ParkerMc** [comment]
I say keep the food blocks. With the crushed ender make some kind of teleport that uses it and can tp you to the cords you put in.

**Comment by CircuitLord** [comment]
I've told you before that we can remove the soups. They were more of a practice for me when I was starting out modding. I don't mod anymore though, but if you want to remove them, that's absolutely fine.

**Comment by Kalman98** [comment]
Okay... but I kind of wanted opinions on everything.

**Comment by pianoman373** [comment]
A thought would be to use the git flow workflow in that nobody can ever push directly to the repository, all changes made are done on a local branch and once the feature is done a pull request is made. If we were to leave all the base-code in the mod but remove all blocks/items we could have a fresh slate to work with and if the creator of a certain feature believes it should be re-added they can do a pull request and/or enhance the feature along the way before it gets merged with the main branch.

This helps keep halfway finished features from sticking around in the mod, but it might also make adding a feature a bit more daunting because you need to finish the whole thing to have anything to show for it.

We also may want to consider somewhat having a theme to the mod, because other completely themeless mods already exist, any ideas we come up with (such as the trash can) will have already been created in similar mods. It doesn't have to be a super theme-centered mod such as IndustrialCraft, but just something that loosely keeps you from having soda machines and potions in the same mod. Even though minecraft itself kind of breaks the rules of theme where we seem to have technology with redstone and also seem to be in a medieval era, if we wanted that kind of freedom we would really want to make the mod as vanilla-like as possible. But that mod has already been done: [https://mods.curse.com/mc-mods/minecraft/243121-quark](url) 

As for a general theme suggestion. I think the least restrictive theme to go with would be something along the lines of rpg-ish with a bit more of a focus on mobs and uncraftable loot. Terraria and Starbound both kind of do this and they both have their own themes but are able to have the most variety of sometimes random content.

**Comment by Kalman98** [comment]
Yes, I was thinking about using a workflow like that. Everyone can create branches for their projects, and when they are ready, they will submit a pull request. The pull request is not so the code can get "approved" by one person, but so that everyone can see it and discuss if everything is optimal, if there are any problems, and everything like that. Then it will be merged with the main branch.

I'm not entirely sure about starting from scratch. It's a nice idea, but a lot of work, and I have a feeling that nobody would want to help rewrite it. As for the mod theme... well, what you said about everything being done is not entirely true. Bridge blocks haven't been done, I've never seen anything quite like crushed ender, nor walking sticks (similar things, but still different), nor even the boxing gloves (which I'm not sure you've had a chance to look at the mechanics behind yet).

I really don't think an RPG theme is the right way to go. There are more than enough mods like that, and it would very quickly get boring. Yes, there are plenty of random mods out there. But that's the point of a random mod: it's got random things. We can make things nobody else has, and if we keep the random theme, we won't be restricted, either. However, I have an idea to make things a little more cohesive. What if we worked on making all of the features of the mod integrate with each other more?

At the moment I can't think how we could accomplish this. For a mod like BuildCraft or ThaumCraft, they have their own energy systems which almost everything uses, thus everything affects everything else in some way. That makes the mod seem more complete, because things lead into other things and you don't end up making two items and never touching the mod again. If we could come up with a unique way to combine everything in the mod, I think it would go to the next level of coolness.

Several things in KalStuff require ender pearls/ender related things. Ender related systems have been more than done before, so that is definitely not ideal. Redstone systems are done, energy systems are done, and the few other "systems" that exist are based on such obscure reasoning that they are pointless (I think I saw a porkchop fueled mod once).

Now, I don't mean to say that KalStuff needs an energy system. I feel like the opposite is true. I've made my example now of how other mods hold together. Maybe we should try to come up with some way to make everything be related to other things... but still not become restricted by a theme? I think it will require some thought.

**Comment by pianoman373** [comment]
Well when I said RPG I didn't really mean RPG. I know there are a few RPG mods out there but what I was thinking would be very different from those. I guess what I might mean is immersion, still not sure if that's the right word. What I was kind of going for was the amount of unique items are in Terraria and Starbound, most of which are uncraftable. This is kind of what drives exploration in these games because you could find something that you couldn't have made yourself.

So it kind of boils down to the fact that if the game had more uncraftable items (and a better system for obtaining them) we would spend a lot more time exploring instead of just farming and mining all the time living in the same square kilometer area 99% of the time. It could kind of be summed up as an anti-skyblock mentality. Spend less time creating massive farms and never-endingly boosting your OP'ness and actually go exploring for unique items.

I can already see how most of the things in the mod could be used in this theme. Walking sticks (obviously) make it easier to travel long distances. Moon flowers can be moved to their own unique biome which other uses can be added do as I suggested. All the ideas we currently have for mobs would fit well with this. Lutes and boxing gloves could also be uncraftable items you get from loot/chest/whatever along with several other non-essential, but still cool-enough-to-go-looking-for items. Such as how in starbound you could find a weapon in some loot that has some special ability associated with it, while there are other versions of that weapon, all their abilities are slightly different and you may favor one over the other. This also brings more of a need for players to want to trade each other's unique items because what the other person has just might be more their style (Such as boxing gloves, not everyone would want to use them rather than some other unique item used for combat).

This is just my current view of how most of the things we want to keep in the mod could be connected to each other through exploration instead of crafting/energy/fuel systems. I'm sure others might come up with a 'theme' or 'system to connect everything' too but I just wanted to throw this idea out there.

TL;DR My idea of a fitting, but not limiting theme would be to add more 'unique items' to the game with a focus on exploration instead of crafting.

**Comment by Kalman98** [comment]
Well, that is an interesting idea, but... I don't really agree with it. One of the most common things a new Minecraft player will want to do is ride a horse. When they ask how to ride a horse, the answer is to put a saddle on it. When they ask how to craft a saddle, the answer is... for absolutely no reason whatsoever you are unable to make your own saddle and must typically go exploring for hours underground trying to find some ruins where you can find a saddle in perfect condition which has no explanation for even being underground, especially in a mine, in the first place. That is super annoying to find out and then have to do.

Boxing gloves are a piece of leather wrapped around a piece of wool, so the concept is very similar. Why would you not be able to craft one? And if we make a lot of items like that, the player will continually have to wonder, "Why on earth can't I make this myself? Why do I have to spend hours on end hoping to get lucky and find this item randomly generated in a chest?"

I don't think the idea of random, uncraftable loot is bad, of course. Record disks are something you definitely could not make yourself, and although dungeons are a very questionable place to find one, it does make slightly more sense than "I just recorded this new CD in the wild by myself with absolutely no equipment for it." So I think we could definitely move _some_ items to randomly generate and be otherwise unobtainable. We just have to be very careful about what they are and why they are like that. And we definitely cannot do that for everything.

It could also be feasible to add a few new structures, if they were carefully thought out, where specific items can only be found, and only rarely. For instance, what about the ruins of a minstrel's guild where there is a very small chance of obtaining a lute? The obvious objection to that, of course, is to question why there would be a minstrel's guild in the middle of the wild. So don't take the example seriously, but that is a thought for something we could also possibly do.

Putting moon flowers in their own biome could work, but we would really need to put a lot of work into that biome to make it worth having at all. I'm not entirely sure what you mean by all of the ideas we have for mobs. I know you put something about woodbeasts in the README, but I don't know what that is. And my final (unrelated) note here: boxing gloves are more meant as a sort of minigame item, and less of a combat item. They really can't be used for combat except under very, very special circumstances, as the whole point of them is that they cannot kill.

**Comment by Kalman98** [comment]
I should probably mention that I do agree that we need a new name. But I don't think we should decide on that until we figure out what we are doing with the mod.

Personally I am starting to lean towards a complete rewrite. I wouldn't mind doing this myself, as I wrote most of the original code and it wouldn't take me too long to finish. But if we were going to do that, we do still have to decide what stays and what goes. I don't even know why round shields and locked chests are still in the code, so those can be removed, but all of the other things we've discussed above need to be decided on.

Maybe instead of deciding what to remove, we should only decide on what to keep. That would make things easier, I think. I would vote on:

walking sticks
bridge blocks
crushed ender
boxing gloves
grapes/grape vines/great grapes/wine
bacon wands(and coffee, for the recipe)
daggers
lutes(when finished)
ender/blaze/food blocks
moon flowers
tea(but not to start, I want to add it from scratch later the way I originally planned)

So if everyone thinks a restart of the mod is a good way to go, could you try to decide on what you think we should keep? Once we figure that out, we can move on to a theme or something, some way for the mod to make more sense. And once we figure _that_ out, we can figure out a name. Then we can make a fancy new trailer and be famous. :smile:

**Comment by Videonauth** [comment]
Regarding the Chicken Nests I can say, I just installed your mod because of them, not having much need of the other recipes in my 150+ mod-pack which I use private. Using the To Many Chickens mod and your Nests should give me a better overview of which chickens are there and which not, this way the hopper function comes as well good into play, just one suggestion why not make the nests themselves act as hoppers?

**Comment by Kalman98** [comment]
@Videonauth actually, thank you for bringing that up. Chicken nests originally _did_ act as hoppers. I'm not entirely sure when, but in one of the updates that functionality got broken and ultimately removed over being fixed. I will look into adding that ability back to them, if I can manage it then it will probably be part of the 0.8.2 bug fixing update, which should be released (relatively) soon.

**Comment by Kalman98** [comment]
And I forgot to mention it in the above comment, but I am glad you have found a use for chicken nests. I think we can leave them in the mod of we have a good reason to. Congratulations, you just saved chicken nests. 👍

**Comment by Videonauth** [comment]
Just tested the nests with the To Many Chickens chickens, sadly the nests don't grab them :-s (I'm not sure how much work this would be to add compatibility to that mod, but I would love to request that) Here the link to the mod in question --> https://minecraft.curseforge.com/projects/too-many-chickens-mod. Actually my chicken house is a big mess with about 40 chickens running around and having hoppers with carpet over them for the whole floor, which is in my eyes semi productive given the fact I not even can enter the house without chickens all the time escaping ;). For me the nests are one of the truly unique tools blocks this mod adds.

**Comment by Kalman98** [comment]
Hm, this is unfortunate. It seems that Soulas125, the author of Too Many Chickens, has not released any source code. The code for chicken nests _should_ automatically detect any new chickens that extend from Vanilla chickens and allow them to be picked up. Seeing as the Too Many Chickens versions don't get picked up, I am thinking that they do not extend the default Chicken class, and as I cannot see the source code, I cannot add any special exceptions to the code, either. It looks like the problem is on Too Many Chickens end, and the only solutions would be to contact Soulas125 and ask them to make the mod open source, to ask them what the base class that all of their chickens extend is (most likely option, I think), or to ask them to rewrite the mod to base all of their code on the default Chicken class (least likely but best option in the long run).

**Comment by Videonauth** [comment]
I'll see what I can do, will contact him for sure about that. He adds a white_egg from which extra white chickens spawn and from the white eggs you can craft all kinds of resource dropping chickens (quite costly recipes as well)

**Comment by ParkerMc** [comment]
Also please open a separate issue since it dose not really fit in this one. Thanks :smile:

**Comment by Kalman98** [comment]
_''_
